cmd/mtbroker/ingress/main.go

```diff
@@ -0,0 +1,196 @@
+	msp := metrics.NewMemStatsAll()
+	msp.Start(ctx, 30*time.Second)
+	if err := view.Register(msp.DefaultViews()...); err != nil {
+		log.Fatalf("Error exporting go memstats view: %v", err)
+	}
+
+	cfg, err := sharedmain.GetConfig(*masterURL, *kubeconfig)
+	if err != nil {
+		log.Fatal("Error building kubeconfig", err)
+	}
+
+	var env envConfig
+	if err := envconfig.Process("", &env); err != nil {
+		log.Fatal("Failed to process env var", zap.Error(err))
+	}
+
+	log.Printf("Registering %d clients", len(injection.Default.GetClients()))
+	log.Printf("Registering %d informer factories", len(injection.Default.GetInformerFactories()))
+	log.Printf("Registering %d informers", len(injection.Default.GetInformers()))
+
+	ctx, informers := injection.Default.SetupInformers(ctx, cfg)
+
+	loggingConfigMapName := cmpresources.MakeCopyConfigMapName(namespaceresources.DefaultConfigMapPropagationName, logging.ConfigMapName())
+	metricsConfigMapName := cmpresources.MakeCopyConfigMapName(namespaceresources.DefaultConfigMapPropagationName, metrics.ConfigMapName())
+
+	loggingConfig, err := cmdbroker.GetLoggingConfig(ctx, "knative-eventing", loggingConfigMapName)
+	if err != nil {
+		log.Fatal("Error loading/parsing logging configuration:", err)
+	}
+	sl, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	logger := sl.Desugar()
+	defer flush(sl)
+
+	logger.Info("Starting the Broker Ingress")
+
+	// Watch the logging config map and dynamically update logging levels.
+	configMapWatcher := configmap.NewInformedWatcher(kubeclient.Get(ctx), "knative-eventing")
+	// Watch the observability config map and dynamically update metrics exporter.
+	updateFunc, err := metrics.UpdateExporterFromConfigMapWithOpts(metrics.ExporterOptions{
+		Component:      component,
+		PrometheusPort: defaultMetricsPort,
+	}, sl)
+	if err != nil {
+		logger.Fatal("Failed to create metrics exporter update function", zap.Error(err))
+	}
+	configMapWatcher.Watch(metricsConfigMapName, updateFunc)
+	// TODO change the component name to broker once Stackdriver metrics are approved.
+	// Watch the observability config map and dynamically update request logs.
+	configMapWatcher.Watch(loggingConfigMapName, logging.UpdateLevelFromConfigMap(sl, atomicLevel, component))
+
+	bin := tracing.BrokerIngressName(tracing.BrokerIngressNameArgs{
+		Namespace:  "knative-eventing",
+		BrokerName: "cluster",
+	})
+	if err = tracing.SetupDynamicPublishing(sl, configMapWatcher, bin,
+		cmpresources.MakeCopyConfigMapName(namespaceresources.DefaultConfigMapPropagationName, tracingconfig.ConfigName)); err != nil {
+		logger.Fatal("Error setting up trace publishing", zap.Error(err))
+	}
+
+	httpTransport, err := cloudevents.NewHTTPTransport(cloudevents.WithBinaryEncoding(), cloudevents.WithMiddleware(pkgtracing.HTTPSpanMiddleware))
+	if err != nil {
+		logger.Fatal("Unable to create CE transport", zap.Error(err))
+	}
+
+	// Liveness check.
+	httpTransport.Handler = http.NewServeMux()
```

Comment:
We should also support passing in the port to listen on here, which should be `httpTransport.Port`